packages/app-types/index.d.ts

```diff
@@ -32,6 +32,84 @@ export interface FirebaseAppConfig {
+   *
+   * Once the token has been properly verified then invoking getAuth() will attempt to
+   * automatically sign in a user with the provided Auth ID Token.
+   *
+   * If the token fails verification then a warning is logged and Auth SDK will not
+   * attempt to sign in a user upon its initalization.
+   */
+  authIdToken?: string;
+
+  /**
+   * An optional AppCheck token.
+   *
+   * If provided, the FirebaseServerApp instance will work to validate the token. The
+   * result of the validation can be monitored by invoking the
+   * FirebaseServerApp.appCheckTokenVerified(). Awaiting the Promise returned by
+   * appCheckTokenVerified is highly recommended if an AppCheck token is provided.
+   *
+   * If the token has been properly verified then the AppCheck token will be
+   * automatically used by Firebase SDKs that support App Check.
+   *
+   * If the token fails verification then a warning is logged and the token will not
+   * be used.
+   */
+  appCheckToken?: string;
+
+  /**
+   * An optional Installation Auth token.
+   *
+   * If provided, the FirebaseServerApp instance will work to validate the token. The
+   * result of the validation can be monitored by invoking the
+   * FirebaseServerApp.installationAuthTokenVerified(). Awaiting the Promise returned by
+   * appCheckTokenVerified is highly recommended before initalization any other Firebase
+   * SDKs.
+   *
+   * If the token has been properly verified then the Installation Auth token will be
+   * automatically used by Firebase SDKs that support Firebase Installations.
+   *
+   * If the token fails verification then a warning is logged and the token will not
+   * be used.
+   */
+  installationAuthToken?: string;
+
+  /**
+   * An optional object. If provided, the Firebase SDK will use a FinalizationRegistry
+   * object to monitor its GC status. The Firebase SDK will cleanup and
+   * delete the corresponding FirebaseServerApp instance when the provided object is
+   * deleted by the garbage collector.
+   *
+   * The intent of this field is to help reduce memory overhead for long-running cloud
+   * functions executing SSR fulfillment without the customer's app needing to
+   * orchestrate FirebaseServerApp cleanup.
+   *
+   * For instance, in the case that asynchronous operations makes it difficult to
+   * determine the finality of the server side rendering pass.
+   *
+   * If the object is not provided then the application must clean up the
+   * FirebaseServerApp instance through it's own standard mechanisms by invoking
+   * deleteApp.
+   *
+   * If the app uses provides an object but uses a JavaScript engine that predates the
+   * support of FinalizationRegistry (introduced in node v14.6.0, for instance), then the
+   * Firebase SDK will not be able to automatically clean up the FirebaseServerApp
+   * instance and an error will be thrown.
+   */
+  deleteOnDeref?: object;
+
+  name: '';
```

Comment:
`name: undefined` should work here

Reply:
Done!

Reply:
Oh i see, i was confusing this with FirebaseServerAppSettings. This should be fine. Carry on.